src/dev/amdgpu/pm4_packet_processor.cc

```diff
@@ -664,7 +664,37 @@ PM4PacketProcessor::unmapQueues(PM4Queue *q, PM4UnmapQueues *pkt)
                 delete mqd;
             }
         }
-        gpuDevice->deallocateAllQueues();
+        gpuDevice->deallocateAllQueues(true);
+        } break;
+      case 3: {
+        auto &hsa_pp = gpuDevice->CP()->hsaPacketProc();
+        for (auto iter : gpuDevice->getUsedVMIDs()) {
+            for (auto id : iter.second) {
+                assert(queues.count(id));
+
+                // Do not unmap KMD queues or static queues
+                if (queues[id]->privileged() || queues[id]->isStatic()) {
```

Comment:
It feels like a better solution to have a helper function both call instead of duplicating this code twice with one minor difference, but it also isn't wrong this way.

Reply:
Good idea. I think the only difference is the bool value sent to the device. Will have to get to this on Friday

Reply:
done